bpf/lib/identity.h

```diff
@@ -9,7 +9,7 @@
 
 static __always_inline bool identity_is_remote_node(__u32 identity)
 {
-	return identity == REMOTE_NODE_ID;
+	return identity == REMOTE_NODE_ID || identity == KUBE_APISERVER_ID;
```

Comment:
Good point, this needs more explanation.

`KUBE_APISERVER_ID` is the fixed identity that means "remote-node" and "kube-apiserver". The intent of this function is to decide which identities represent other k8s nodes in the cluster so that the appropriate routing decisions and so on. If the apiserver is actually running on a non-k8s node, then this identity will not be used and that case will not be hit here, and furthermore this is correct as we shouldn't be making "in-cluster" routing decisions based on an apiserver that's outside the cluster.

This is necessary more because we're changing the identity of nodes from the perspective of the local agent, not because of the policy pieces; The policy side is handled through the standard policymap infrastructure.

Reply:
Oh! Ok. Could we rename `KUBE_APISERVER_ID` to `KUBE_APISERVER_NODE_ID` or `LOCAL_KUBE_APISERVER_ID` to clarify that it's only for nodes running kube-apiserver backend pods?

Reply:
:+1: I think I'll avoid "LOCAL" in the naming since this is indicating a remote node, but KUBE_APISERVER_NODE_IDENTITY seems reasonable.  (In general I try to ensure that any reference to security identities use the full word 'identity' since 'id' is a common short form for other identifications such as endpoint ids which are unrelated to security identity).

Reply:
Now I look back over this code, I see that I used `_ID` as the suffix here for consistency with the other identities. I guess I'll just keep it consistent with the _ID suffix.